plutus-core/plutus-core/examples/PlutusCore/Examples/Builtins.hs

```diff
@@ -418,12 +443,48 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni ExtensionFun where
     -- See Note [Versioned builtins]
     toBuiltinMeaning ver ExtensionVersion =
         makeBuiltinMeaning
-        @(() -> EvaluationResult Integer)
-        (\(_ :: ()) -> EvaluationSuccess $ case ver of
-                ExtensionFunV0 -> 0
-                ExtensionFunV1 -> 1
-        )
-        mempty  -- Whatever
+            @(() -> EvaluationResult Integer)
+            (\(_ :: ()) -> EvaluationSuccess $ case ver of
+                    ExtensionFunV0 -> 0
+                    ExtensionFunV1 -> 1)
+            (\_ _ -> ExBudgetLast mempty)  -- Whatever
+
+    toBuiltinMeaning _ TrackCosts = unsafePerformIO $ do
+        counterVar <- newMVar 0
+        resultsVar <- newIORef []
+        let finalize =
+                tryTakeMVar counterVar >>= \case
+                    Nothing      -> void $ atomicModifyIORef'_ resultsVar (0 :)
+                    Just counter -> do
+                        _ <- atomicModifyIORef'_ resultsVar (counter :)
+                        putMVar counterVar 0
+
+            regBudget budget = do
+                counter <- takeMVar counterVar
+                let counter' = succ counter
+                putMVar counterVar counter'
+                addFinalizer budget finalize
+                when (counter' >= 100) performMinorGC
+
+            regBudgets (ExBudgetLast budget)         = do
+                regBudget budget
+                pure $ ExBudgetLast budget
+            regBudgets (ExBudgetCons budget budgets) = do
+                regBudget budget
+                budgets' <- unsafeInterleaveIO $ regBudgets budgets
+                pure $ ExBudgetCons budget budgets'
+
+            linear1 = ModelOneArgumentLinearCost $ ModelLinearSize 1 1
+            model   = CostingFun linear1 linear1
+        pure $ makeBuiltinMeaning
+            @(Data -> [Integer])
+            (\_ -> unsafePerformIO $ do
+                performMinorGC
+                threadDelay 50000  -- Giving GC enough time to finish up.
```

Comment:
Right, silly me. However that `threadDelay 50000` does affect the results: without it some finalizers are run too late and we don't get some of the trailing elements of the stream registered. It's like 50 elements on my machine, but still.

I just tried `threadDelay 1`, it helped just as much. So I also tried `yield` and it didn't help with the loss of trailing elements. I don't have the slightest idea of what's going on, but it's not too important anyway. I'll make `threadDelay 1` just to please the gods of GHC, so that they don't run finalizers prematurely (and maybe don't insert unwanted join points into my code so much).

Reply:
I've changed the logic of the builtin and it now works better and tells us more without any `threadDelay` garbage. Documented all of it.